feat : login activity mvvm migration

gururani-abhishek opened the pull request:
MOBILE-67

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything
- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.


- [x] If you have multiple commits please combine them into one commit by squashing them.

gururani-abhishek committed:
feat : login activity mvvm migration

gururani-abhishek committed:
feat : login activity mvvm migration

-> modified Ui state class
-> added unit tests for LoginViewModel

gururani-abhishek committed:
feat : login activity mvvm migration

-> updated LoginViewModel unit tests
-> added unit tests for functions related to login feature in UserAuthRepositoryTest

jawidMuhammadi commented on app/src/main/java/org/mifos/mobile/repositories/UserAuthRepository.kt:
```diff
@@ -15,4 +18,17 @@ interface UserAuthRepository {
+
+    fun login(username: String, password: String): Observable<User?>?
+
+    fun loadClient() : Observable<Page<Client?>?>?
+
+    fun saveAuthenticationTokenForSession(user: User)
+
+    fun reInitializeService()
+
+    fun setClientId(clientId: Long?)
```
It is recommended to avoid having these functions within the UserAuthRepository class as it goes against the Single Responsibility Principle (SRP). To address this concern, it is great to create a new interface called `ClientRepository` along with its implementation, `ClientRepositoryImp`. This new class can then be used in the `LoginViewModel`.

gururani-abhishek replied:
Got it, so insightful. made the req changes. 😃

gururani-abhishek committed:
feat : login activity mvvm migration

-> created ClientRepository, its implementation and unit tests.
-> removed Client related methods from UserAuthRepository, and its unit tests.
-> injected ClientRepositoryImp to LoginViewModel through Hilt, and added its entry to repository module of DI

gururani-abhishek committed:
feat : login activity mvvm migration

-> injected preferences helper dependency through hilt

gururani-abhishek committed:
Merge remote-tracking branch 'origin/mvvm_architecture' into login-activity-mvvm-migration

# Conflicts:
#	app/src/main/java/org/mifos/mobile/injection/module/RepositoryModule.kt
#	app/src/main/java/org/mifos/mobile/repositories/ClientRepository.kt
#	app/src/main/java/org/mifos/mobile/repositories/ClientRepositoryImp.kt
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepository.kt
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepositoryImp.kt
#	app/src/test/java/org/mifos/mobile/repositories/ClientRepositoryImpTest.kt

gururani-abhishek commented:
Hey @jawidMuhammadi, do have a look at this whenever you find the bandwith. I've resolved the conflicts as well. 😃

gururani-abhishek committed:
Merge remote-tracking branch 'origin/mvvm_architecture' into login-activity-mvvm-migration

# Conflicts:
#	app/src/main/java/org/mifos/mobile/injection/module/RepositoryModule.kt
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepository.kt
#	app/src/main/java/org/mifos/mobile/repositories/UserAuthRepositoryImp.kt
#	app/src/test/java/org/mifos/mobile/repositories/UserAuthRepositoryImpTest.kt
